.github/workflows/regression.yml

```diff
@@ -45,11 +47,30 @@ env:
 
 jobs:
 
+  # Determine if the suite should run
+  check-should-run:
+    runs-on: ubuntu-latest
+    outputs:
+      should-run: ${{ steps.set-should-run.outputs.SHOULD_RUN }}
+    steps:
+      - name: Determine if should run
+        id: set-should-run
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "SHOULD_RUN=false" >> $GITHUB_OUTPUT
+          elif [[ "${{ github.event_name }}" != "pull_request_review" &&
```

Comment:
Is this intended to match the `workflow_call`? Or are there any other events that should match?

Reply:
Thank you for this comment, it made me think a bit more about the whole check 🤔 

Updates in [this commit](https://github.com/vectordotdev/vector/pull/21507/commits/1547502b6987db491da34b8ae55d76cff99d6f03):
* Lifted the check `github.event_name != 'pull_request'` from the previously named job `should-run` (now called `check-source-changed`)
* Added check for the new label
  * caveat we still want this suite to be triggered by review comments hence the check for `pull_request_review`
* All other events fall under the `else` block, and the `should-run` output becomes `true`.

A couple of checks:
* https://github.com/vectordotdev/vector/pull/21507#pullrequestreview-2373537622
* https://github.com/vectordotdev/vector/pull/21507#pullrequestreview-2373568988
(I manually cancelled them after they progressed)

Reply:
I'm having a hard time following the conditional logic in here 😓 I think this might be more readable if we did something like (might contain typos):

```
if [[ "${{ github.event_name }}" == "pull_request" ]]; then
  echo "SHOULD_RUN=false" >> $GITHUB_OUTPUT
elif [[ "${{ github.event_name }}" == "pull_request_review" ]]; then
  echo "SHOULD_RUN=true" >> $GITHUB_OUTPUT
elif [[ "${{ contains(github.event.pull_request.labels.*.name, 'ci-condition: skip-regression') }}" == "true" ]]; then
  # The small caveat here is that we still want to run the suite via a review comment.
  echo "SHOULD_RUN=false" >> $GITHUB_OUTPUT
else
  echo "SHOULD_RUN=true" >> $GITHUB_OUTPUT
fi
```

I think these simplified expressions make it easier to see the intent.

Reply:
I saw this just now after merging. I think these are equivalent. I wanted to see how this behaves for now dependabot PRs. I can follow-up to clean it up even more.

One question, for the following:
```
elif [[ "${{ github.event_name }}" == "pull_request_review" ]]; then echo "SHOULD_RUN=true" >> $GITHUB_OUTPUT
```

I suppose we will never hit this unless this was triggered by a `ci-run-regression` comment?

Reply:
Agreed, I think they are equivalent, I just find it hard to understand the current conditions 😓 



> One question, for the following:
> 
> ```
> elif [[ "${{ github.event_name }}" == "pull_request_review" ]]; then echo "SHOULD_RUN=true" >> $GITHUB_OUTPUT
> ```
> 
> I suppose we will never hit this unless this was triggered by a `ci-run-regression` comment?

Actually, I think `pull_request_review` will never be the trigger. Instead, it'll be `workflow_call` (where the caller, [ci-review.yml](https://github.com/vectordotdev/vector/blob/master/.github/workflows/ci-review-trigger.yml) has the `pull_request_review` trigger). So instead this should be:

```
elsif [[ "${{ github.event_name }}" == "workflow_call" ]]; then echo "SHOULD_RUN=true" >> $GITHUB_OUTPUT
```

Reply:
Closing the loop: https://github.com/vectordotdev/vector/pull/21540